Documentation/upgrade.md

```diff
@@ -238,16 +291,36 @@ Each of these replica sets will need to be edited to update the image version, t
 
 In this example, we are going to walk through the case where `useAllNodes: true` was set in the cluster spec, so there will be a single daemon set managing all the OSD pods.
 
-Let's edit the image version of the pod template in the daemon set:
-```bash
-kubectl -n rook set image daemonset/rook-ceph-osd rook-ceph-osd=rook/rook:v0.5.1
+In addition to updating the version of the container, we will need to add a new environment variable for the `uid` of the cluster. First, we will look up the value of the cluster `uid` that is required for the new variable.
+```
+kubectl -n rook get cluster rook -o jsonpath='{.metadata.uid}'
+```
+
+Now let's edit the daemonset or replicaset (depending on how the OSDs were deployed).
+```
+# If using a daemonset for all nodes
+kubectl -n rook edit daemonset rook-ceph-osd
+
+# If using a replicaset for specific nodes
+kubectl -n rook edit replicaset rook-ceph-osd-<node>
+```
+
+Add two new lines next to other environment variables, replacing the example value with your cluster `uid`:
+```
+        - name: ROOK_CLUSTER_ID
+          value: 6a4d0134-1800-11e8-b483-08002732be0b
```

Comment:
instead of a real uid value here, perhaps put a `<your_uid>` placeholder here instead so people don't get too copy/paste trigger happy? from my suggestion above, we would already show them what an example uid looks like from the look up output